clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -222,6 +223,27 @@ public void clearPartitionRecords() {
         partitionSizes();
         data = null;
     }
+    
+    public static void validateProducerIds(short version, ProduceRequestData data) {
+        if (version >= 3) {
```

Comment:
For the uninitiated, can we move and/or comment why version `3` is special?

Reply:
In producerequest.json:
>   // Version 3 adds the transactional ID, which is used for authorization when attempting to write
  // transactional data.  Version 3 also adds support for Kafka Message Format v2.
  
 Basically before version 3, these fields aren't used.